virttest/libvirt_xml/vm_xml.py

```diff
@@ -1472,6 +1472,42 @@ def remove_all_boots(self):
             pass  # Element already doesn't exist
         self.xmltreefile.write()
 
+    def set_boot_order_by_target_dev(self, target, order):
+        """
+        Set boot order by target dev
+
+        :param target: The target dev on host
+        :param order: The boot order number
+        """
+        devices = self.get_devices()
+        for device in devices:
+            if device.device_tag == "disk":
+                if device.target.get("dev") == target:
+                    device.boot = order
+        self.set_devices(devices)
+
+    def set_os_attrs(self, **attr_dict):
+        """
+        Set attributes of VMOSXML
+
+        :param attr_dict: The key words of os attributes
+        """
+        try:
+            os_xml = getattr(self, "os")
+            if attr_dict:
+                for name, value in attr_dict.items():
+                    setattr(os_xml, name, value)
+            setattr(self, "os", os_xml)
```

Comment:
Also here, use self.os = os_xml

Reply:
same comments as above

Reply:
here should be no problem for syntax check as the check fail because the definition after attr get, anyway, it is also a method to set attr, so it's ok